### PR TITLE
Update transaction related data type for CKB v0.19.0

### PR DESCRIPTION
**Pull request by duanyytop**
# CKB v0.19.0 changes

### Feature

* Update transaction related data type
* Update genesis block system cell data hash and cell dep
* Remove `epoch_reward` of Epoch
* Remove `seal` of Header
* Update `cell_dep -> dep_type` and `script -> hash_type`  data type

### Test

* Update wallet test case

### CODE REVIEW

File: console/src/main/java/org/nervos/ckb/Wallet.java [hunk 1]
```diff
@@ -26,21 +28,29 @@
...
   private String privateKey;
+  private Script lockScript;
+  private SystemScriptCell systemScriptCell;
   private CKBService ckbService;
 
-  public Wallet(String privateKey, Script lockScript, String nodeUrl) {
+  public Wallet(String privateKey, String nodeUrl) {
     this.privateKey = privateKey;
-    this.lockScript = lockScript;
+    HttpService.setDebug(false);
```
Comment (shaojunda) [inline console/src/main/java/org/nervos/ckb/Wallet.java]: what is the effect of this method?

File: core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package org.nervos.ckb.methods.type.cell;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.nervos.ckb.methods.type.OutPoint;
+
+/** Copyright © 2019 Nervos Foundation. All rights reserved. */
+public class CellDep {
+
+  public static final String CODE = "code";
+  public static final String GROUP = "group";
```
Comment (ashchan) [inline core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java]: Isn’t that dep_group?

Comment (ashchan) [inline core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java]:
```suggestion
  public static final String DEP_GROUP = "dep_group";
```

File: core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  @JsonProperty("out_point")
+  public OutPoint outPoint;
+
+  @JsonProperty("dep_group")
```
Comment (ashchan) [inline core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java]:
```suggestion
  @JsonProperty("dep_type")
```

File: core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  public String depGroup;
```
Comment (ashchan) [inline core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java]:
```suggestion
  public String depType;
```

File: core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  public CellDep() {}
+
+  public CellDep(OutPoint outPoint, String depGroup) {
```
Comment (ashchan) [inline core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java]:
```suggestion
  public CellDep(OutPoint outPoint, String depType) {
```

File: core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    this.outPoint = outPoint;
+    this.depGroup = depGroup;
```
Comment (ashchan) [inline core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java]:
```suggestion
    this.depType = depType;
```

File: core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java (hunk 2 above)
Comment (shaojunda) [inline core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java]: the latest version is dep_group

File: console/src/main/java/org/nervos/ckb/Wallet.java [hunk 7]
```diff
@@ -54,28 +64,24 @@ private Transaction generateTx(List<Receiver> receivers) throws Exception {
...
     List<CellOutput> cellOutputs = new ArrayList<>();
     AddressUtils addressUtils = new AddressUtils(Network.TESTNET);
     for (Receiver receiver : receivers) {
       String blake2b = addressUtils.getBlake160FromAddress(receiver.address);
       cellOutputs.add(
           new CellOutput(
               receiver.capacity.toString(),
-              "0x",
-              new Script(systemScriptCell.cellHash, Arrays.asList(blake2b))));
+              new Script(systemScriptCell.cellHash, Collections.singletonList(blake2b))));
```
Comment (shaojunda) [inline console/src/main/java/org/nervos/ckb/Wallet.java]: what is the value of hash_type here?

File: console/src/main/java/org/nervos/ckb/Wallet.java (hunk 1 above)
Comment (duanyytop) [reply console/src/main/java/org/nervos/ckb/Wallet.java]: lockScript can generate from private and system script, so I remove it.

Add `HttpService.setDebug(false)` to make http log more clear.

File: core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java (hunk 2 above)
Comment (duanyytop) [reply core/src/main/java/org/nervos/ckb/methods/type/cell/CellDep.java]: Have changed.

File: console/src/main/java/org/nervos/ckb/Wallet.java (hunk 7 above)
Comment (duanyytop) [reply console/src/main/java/org/nervos/ckb/Wallet.java]: OK. I will fix it.

File: console/src/main/java/org/nervos/ckb/Wallet.java [hunk 8]
```diff
@@ -127,11 +141,26 @@ private CellInputs getCellInputs(String lockHash, BigInteger needCapacities) thr
     return new CellInputs(cellInputs, new BigDecimal(inputsCapacities).toBigInteger());
   }
 
-  class CellInputs {
+  private Script generateLockScript(String privateKey, String codeHash) {
+    String publicKey = Sign.publicKeyFromPrivate(Numeric.toBigInt(privateKey), true).toString(16);
+    String blake160 =
+        Numeric.prependHexPrefix(Numeric.cleanHexPrefix(Hash.blake2b(publicKey)).substring(0, 40));
+    return new Script(codeHash, Collections.singletonList(blake160), Script.TYPE);
```
Comment (shaojunda) [inline console/src/main/java/org/nervos/ckb/Wallet.java]: can I generate a script which hash_type is data?

Comment (duanyytop) [reply console/src/main/java/org/nervos/ckb/Wallet.java]: Yes. Last parameter is hash_type, anyone could set value.

File: core/src/main/java/org/nervos/ckb/methods/response/CkbLockHash.java [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package org.nervos.ckb.methods.response;
+
+import org.nervos.ckb.methods.Response;
+
+/** Copyright © 2019 Nervos Foundation. All rights reserved. */
+public class CkbLockHash extends Response<String> {
```
Comment (classicalliu) [inline core/src/main/java/org/nervos/ckb/methods/response/CkbLockHash.java]:
```suggestion
public class CkbScriptHash extends Response<String> {
```

Is this script hash?

Comment (duanyytop) [reply core/src/main/java/org/nervos/ckb/methods/response/CkbLockHash.java]: OK. It is lock hash and also script hash.

Comment (duanyytop) [reply core/src/main/java/org/nervos/ckb/methods/response/CkbLockHash.java]: Script hash may be more general.

**Pull request merged**